packages/react/src/components/Popup/Popup.tsx

```diff
@@ -124,30 +124,32 @@ function useHideSiblingNodes(props: IPopupProps, root: React.RefObject<HTMLDivEl
   const isModalOrPanel = props['aria-modal'];
 
   React.useEffect(() => {
-    const targetDocument = getDocument();
-    if (isModalOrPanel && targetDocument && root && root.current) {
-      const popupPortalNode = root.current.parentElement?.parentElement;
-      let nodesToHide: HTMLElement[] = findSiblingNodes(popupPortalNode, popupPortalNode?.parentElement);
-
-      //if popupPortalNode is not a direct child of body, its ancestor's siblings need to be hidden as well.
-      if (popupPortalNode?.parentElement !== targetDocument.body) {
-        const popupAncestorNode = findAncestorNode(root.current, targetDocument);
-        nodesToHide.concat(findSiblingNodes(popupAncestorNode, targetDocument.body));
-      }
+    if (props.enableAriaHiddenSiblings) {
+      const targetDocument = getDocument();
+      if (isModalOrPanel && targetDocument && root && root.current) {
+        const popupPortalNode = root.current.parentElement?.parentElement;
+        let nodesToHide: HTMLElement[] = findSiblingNodes(popupPortalNode, popupPortalNode?.parentElement);
+
+        //if popupPortalNode is not a direct child of body, its ancestor's siblings need to be hidden as well.
+        if (popupPortalNode?.parentElement !== targetDocument.body) {
+          const popupAncestorNode = findAncestorNode(root.current, targetDocument);
+          nodesToHide.concat(findSiblingNodes(popupAncestorNode, targetDocument.body));
```

Comment:
I know you didn't change this line (aside from nesting it in a conditional) but I think there is a bug here as [`Array.concat()` does not modify the existing arrays](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) so the ancestor's siblings will never be hidden.

I believe this should be changed to:

```ts
nodesToHide = nodesToHide.concat(findSiblingNodes(popupAncestorNode, targetDocument.body));
```

Reply:
Nice job spotting this. I'm sitting on this PR still to see where the discussion [on the linked issue](https://github.com/microsoft/fluentui/issues/21091) goes, as perhaps we can address this issue in a cleaner way.

Reply:
I've made this change.